app/src/main/java/com/kickstarter/features/search/viewmodel/SearchAndFilterViewModel.kt

```diff
@@ -1,90 +1,175 @@
 import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.plus
 import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.text.isNotBlank
 
 data class SearchUIState(
     val isLoading: Boolean = false,
-    val isErrored: Boolean = false,
-    val popularProjectsList: List<Project> = emptyList(), // TODO MBL-2135 popular & search lists could be potentially unified
+    val popularProjectsList: List<Project> = emptyList(),
     val searchList: List<Project> = emptyList()
 )
 
+@OptIn(FlowPreview::class)
 class SearchAndFilterViewModel(
     private val environment: Environment,
     private val testDispatcher: CoroutineDispatcher? = null
 ) : ViewModel() {
 
     private val scope = viewModelScope + (testDispatcher ?: EmptyCoroutineContext)
     private val apolloClient = requireNotNull(environment.apolloClientV2())
+    private val analyticEvents = requireNotNull(environment.analytics())
 
     private val _searchUIState = MutableStateFlow(SearchUIState())
     val searchUIState: StateFlow<SearchUIState>
         get() = _searchUIState
             .asStateFlow()
             .stateIn(
-                scope = viewModelScope,
+                scope = scope,
                 started = SharingStarted.WhileSubscribed(),
                 initialValue = SearchUIState()
             )
 
-    // Popular projects sorting selection
+    // - Popular projects sorting selection
     private val popularDiscoveryParam = DiscoveryParams.builder().sort(DiscoveryParams.Sort.POPULAR).build()
-    // TODO Will be updated with the params used to call search with, private for now
-    private val listOfSearchParams = listOf(popularDiscoveryParam)
+
+    private val _params = MutableStateFlow(popularDiscoveryParam)
+    val params: StateFlow<DiscoveryParams> = _params
+
+    private val debouncePeriod = 300L
+    private val _searchTerm = MutableStateFlow("")
+    private val searchTerm: StateFlow<String> = _searchTerm
+
+    private var errorAction: (message: String?) -> Unit = {}
+
+    private var projectsList = emptyList<Project>()
+    private var popularProjectsList = emptyList<Project>()
+
+    init {
+        scope.launch {
+            searchTerm
+                .debounce(debouncePeriod)
+                .onEach { debouncedTerm ->
+                    // - Reset to initial state in case of empty search term
+                    if (debouncedTerm.isEmpty() || debouncedTerm.isBlank()) {
+                        _params.emit(popularDiscoveryParam)
+                    } else
+                        _params.emit(
```

Comment:
In future additions, with category selection or project sort, there will be inputs to this VM (similar to `updateSearchTerm`) that will add new emissions to  _params with each user selection.
In the future will likely not collect `searchTerm` chain but rather `params`